lib/ddtrace/configuration/settings.rb

```diff
@@ -65,16 +92,38 @@ def initialize(*_)
           end
         end
 
+        # Internal tracer {Datadog::Statsd} metrics collection.
+        #
+        # The list of metrics collected can be found in {Datadog::Ext::Diagnostics::Health::Metrics}.
         settings :health_metrics do
+          # Enable health metrics collection.
+          #
+          # @default `DD_HEALTH_METRICS_ENABLED` environment variable, otherwise `false`
+          # @return [Boolean]
           option :enabled do |o|
             o.default { env_to_bool(Datadog::Ext::Diagnostics::Health::Metrics::ENV_ENABLED, false) }
             o.lazy
           end
 
+          # {Datadog::Statsd} instance to collect health metrics.
+          #
+          # If `nil`, health metrics creates a new {Datadog::Statsd} client with default agent configuration.
+          #
+          # @default `nil`
+          # @return [Datadog::Statsd,nil] a custom {Datadog::Statsd} instance
+          # @return [nil] an instance with default agent configuration will be lazily created
           option :statsd
         end
 
+        # Tracer startup debug log statement configuration.
         settings :startup_logs do
+          # Enable startup logs collection.
+          #
+          # If `nil`, defaults to logging startup logs when `ddtrace` detects that the application
+          # is *not* running in a development environment.
+          #
+          # @default `DD_TRACE_STARTUP_LOGS` environment variable, otherwise `nil`
+          # @return [Boolean,nil]
           option :enabled do |o|
```

Comment:
Probably one more that could be made generic, although it does somewhat say trace on the env flag :/